components/ILIAS/Chatroom/classes/class.BannedUsersTable.php

```diff
@@ -0,0 +1,170 @@
+            ILIAS_HTTP_PATH . '/' . $this->ctrl->getLinkTargetByClass(ilObjChatroomGUI::class, 'ban-handleTableActions')
+        );
+
+        $url_builder = new UI\URLBuilder($uri);
+        list(
+            $url_builder, $action_parameter_token_copy, $row_id_token
+            ) =
+            $url_builder->acquireParameters(
+                $query_params_namespace,
+                'action',
+                'user_ids'
+            );
+
+        return [
+            'delete' => $this->ui_factory->table()->action()->multi(
+                $this->lng->txt('unban'),
+                $url_builder->withParameter($action_parameter_token_copy, 'delete'),
+                $row_id_token
+            ),
+        ];
+    }
+
+    private function initRecords(): void
+    {
+        if ($this->records === null) {
+            $this->records = [];
+            $i = 0;
+            $entries = $this->banned_users;
+
+            foreach ($entries as $entry) {
+                $this->records[$i]['user_id'] = $entry['user_id'];
+                $this->records[$i]['login'] = $entry['login'];
+                $this->records[$i]['firstname'] = $entry['firstname'];
+                $this->records[$i]['lastname'] = $entry['lastname'];
+                if (is_numeric($entry['timestamp']) && $entry['timestamp'] > 0) {
+                    $this->records[$i]['timestamp'] = ilDatePresentation::formatDate(
+                        new ilDateTime($entry['timestamp'], IL_CAL_UNIX)
+                    );
+                }
+
+                $this->records[$i]['actor'] = $entry['actor'];
+                ++$i;
+            }
+        }
+    }
+
+    public function getRows(
+        UI\Component\Table\DataRowBuilder $row_builder,
+        array $visible_column_ids,
+        Data\Range $range,
+        Data\Order $order,
+        ?array $filter_data,
+        ?array $additional_parameters
+    ): \Generator {
+        $records = $this->getRecords($range, $order);
+
+        foreach ($records as $record) {
+            $row_id = (string) $record['user_id'];
+            yield $row_builder->buildDataRow($row_id, $record);
+        }
+    }
+
+    public function getTotalRowCount(
+        ?array $filter_data,
+        ?array $additional_parameters
+    ): ?int {
+        $this->initRecords();
+
+        return count((array) $this->records);
+    }
+
+    private function sortedRecords(Data\Order $order): array
+    {
+        $records = $this->records;
+        [$order_field, $order_direction] = $order->join([], fn($ret, $key, $value) => [$key, $value]);
+
+        return ilArrayUtil::stableSortArray($records, $order_field, strtolower($order_direction));
+    }
+
+    private function getRecords(Data\Range $range, Data\Order $order): array
+    {
+        $this->initRecords();
+        $records = $this->sortedRecords($order);
+
+        return $this->limitRecords($records, $range);
+    }
+
+    private function limitRecords(array $records, Data\Range $range): array
```

Comment:
Can you try to specify the shape of the array here for both arrays?

`list<array<string, mixed>>`